generators/unifiedsearch-core/src/main/java/uk/gov/justice/services/clients/unifiedsearch/core/UnifiedSearchFileFinder.java

```diff
@@ -0,0 +1,58 @@
+package uk.gov.justice.services.clients.unifiedsearch.core;
+
+import static java.lang.String.format;
+import static java.nio.file.Paths.get;
+import static java.util.Arrays.asList;
+import static java.util.Collections.list;
+
+import java.io.IOException;
+import java.net.URL;
+import java.nio.file.Path;
+import java.util.List;
+
+import javax.enterprise.context.ApplicationScoped;
+
+@ApplicationScoped
+public class UnifiedSearchFileFinder {
+
+    private static final String TRANSFORMER_FILE_PATH = "transformer/";
+    private static final String UNIFIED_SEARCH_DESCRIPTOR = "yaml/unified-search-descriptor.yaml";
+
+    public List<Path> getUnifiedSearchDescriptor() {
+        final URL url = findOnClasspath(UNIFIED_SEARCH_DESCRIPTOR)
+                .stream()
+                .findFirst()
+                .orElseThrow(() -> new UnifiedSearchException("Unable to find file yaml/unified-search-descriptor.yaml on classpath"));
```

Comment:
should we use UNIFIED_SEARCH_DESCRIPTOR constant in the message?

Reply:
done